oshi-core/src/test/java/oshi/util/MemoizerTest.java

```diff
@@ -65,39 +67,59 @@ public void after() throws InterruptedException {
 
     @Test
     public void get() throws Throwable {
+        // Do 20+ seconds of tests with no refresh.
         long iterationDurationNanos = TimeUnit.MILLISECONDS.toNanos(1);
+        long ttlNanos = -1;
         for (int r = 0; r < 20_000; r++) {
-            assertTrue(run(iterationDurationNanos, -1));
+            run(iterationDurationNanos, ttlNanos);
         }
+        // Do 40+ seconds of tests with refresh after 0.1 ms.
         iterationDurationNanos = TimeUnit.MILLISECONDS.toNanos(10);
+        ttlNanos = iterationDurationNanos / 100;
+        assertNotEquals(0, ttlNanos); // avoid div/0 later
         for (int r = 0; r < 4000; r++) {
-            assertTrue(run(iterationDurationNanos, iterationDurationNanos / 100));
+            run(iterationDurationNanos, ttlNanos);
         }
     }
 
-    private boolean run(final long iterationDurationNanos, final long ttlNanos) throws Throwable {
+    private void run(final long iterationDurationNanos, final long ttlNanos) throws Throwable {
```

Comment:
How is this method long? :) Looks like the checker does not ignore lines containing only comments.

Reply:
[Cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) calculation.  15 is the threshold.

Reply:
Are you sure it assesses cyclomatic complexity? This is not obvious from the description of the problem: _When methods are excessively long this usually indicates that the method is doing more than its name/signature might suggest. They also become challenging for others to digest since excessive scrolling causes readers to lose focus. Try to reduce the method length by creating helper methods and removing any copy/pasted code._

Reply:
Not really sure... but can't imagine that it includes comments.  In any case I refactored some bits to make the problem go away ;)

Reply:
Digging into Codacy it's using the legacy PMD 5.x for this rule. I switched to 6.x which clearly states NCSS.